torch/lib/c10d/test/ProcessGroupNCCLTest.cpp

```diff
@@ -172,7 +173,14 @@ class AllreduceNCCLTest : public NCCLTest {
     launchDeviceSleep();
     valueInitialization();
 
-    return pg_->allreduce(tensors_);
+    using namespace torch::autograd::profiler;
+    // Make sure enabling profile does not make any issue. Note, in single
+    // process multi-device mode we do not expect any events be populated for
+    // collective operations.
+    enableProfiler({ProfilerState::CPU});
+    auto results = pg_->allreduce(tensors_);
+    disableProfiler();
+    return results;
```

Comment:
Do we need to add a cpp test since we're already covered by python tests?

Reply:
I think it would be useful from profiler perspective to just test the C++ API as well, which skips the parsing/event aggregation logic in profiler that happens in python. We have similar tests in https://github.com/pytorch/pytorch/blob/master/test/cpp/jit/test_misc.cpp#L2185-L2198